From 04a708330c536d1cb7430c38ffcbaa02dcffe2de Mon Sep 17 00:00:00 2001
From: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>
Date: Sun, 31 Mar 2019 19:55:09 +0000
Subject: [PATCH 2/5] crio.service: allow limited restarts on failure

crio.service may fail to start if network is not fully available.

This change allows crio.service to restart on failure up to 6 times
with a 10 second delay between each attempt, during periods of 2
minutes -following same approach as the one for kubelet.service-
and being applied on top of the previous commit that updated the
ExecStart line to appropriate crio bin path.

Fixes: clearlinux/distribution#192

Signed-off-by: Jose Lamego <jose.a.lamego@intel.com>
Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>
---
 contrib/systemd/crio.service | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/contrib/systemd/crio.service b/contrib/systemd/crio.service
index 896f64f..59d93be 100644
--- a/contrib/systemd/crio.service
+++ b/contrib/systemd/crio.service
@@ -3,12 +3,14 @@ Description=Open Container Initiative Daemon
 Documentation=https://github.com/kubernetes-sigs/cri-o
 Wants=network-online.target
 After=network-online.target
+StartLimitIntervalSec=120
+StartLimitBurst=6
 
 [Service]
 Type=notify
 EnvironmentFile=-/etc/sysconfig/crio
 Environment=GOTRACEBACK=crash
-ExecStart=/usr/local/bin/crio \
+ExecStart=/usr/bin/crio \
           $CRIO_STORAGE_OPTIONS \
           $CRIO_NETWORK_OPTIONS \
           $CRIO_METRICS_OPTIONS
@@ -19,7 +21,8 @@ LimitNPROC=1048576
 LimitCORE=infinity
 OOMScoreAdjust=-999
 TimeoutStartSec=0
-Restart=on-abnormal
+Restart=on-failure
+RestartSec=10
 
 [Install]
 WantedBy=multi-user.target
-- 
2.21.0

